File: sally/leetcode/338_Counting Bits.cpp
// 338. Counting Bits
// Runtime: 8 ms, faster than 33.39% of C++ online submissions for Counting Bits.
// Memory Usage: 8.4 MB, less than 38.35% of C++ online submissions for Counting Bits.

class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> result;
        result.push_back(0);
        for(int i = 1; i <= n; i++){
            int target = i;
            int res = 1;
            while(1){
                if(target == 1) break;
                if(target % 2 == 1) res++;
                target /= 2;
            }
            result.push_back(res);
        }
        return result;
    }
};

File: coodingpenguin/leetcode/338_CountingBits.py
# 문제: 338. Counting Bits
# 링크: https://leetcode.com/problems/counting-bits/


# 시간/공간: 80ms / 21MB
class Solution:
    def countBits(self, n: int) -> List[int]:
        answer = []
        for i in range(n + 1):
            answer.append(bin(i).count("1"))
        return answer
